Tests/libhostmgrTests/BuildkiteScriptBuilderTests.swift

```diff
@@ -97,7 +97,25 @@ class BuildkiteScriptBuilderTests: XCTestCase {
             arguments: ["/Users/my builder user/.bashrc"]
         )
 
-        XCTAssertEqual(#"buildkite-agent bootstrap "/Users/my builder user/.bashrc""#, command.escapedText)
+        XCTAssertEqual(#"buildkite-agent bootstrap '/Users/my builder user/.bashrc'"#, command.escapedText)
+    }
+
+    func testThatCommandQuotesArgumentsContainingSingleQuotes() throws {
+        let command = BuildkiteScriptBuilder.Command(
+            command: "buildkite-agent bootstrap",
+            arguments: ["--name", #"My 'very important' agent"#]
+        )
+
+        XCTAssertEqual("buildkite-agent bootstrap --name 'My '\''very important'\'' agent'", command.escapedText)
```

Comment:
This is interesting ^^

This is totally valid but uses string _concatenation_ (concatenates `My ` with `\'` then with `very important` then with `\'` then with ` agent`) instead of string _interpolation_… would never have considered doing it that way, and not sure why `spm_shellEscaped` decided to go that route here 😅